roles/k8s_cluster_info/tasks/main.yml

```diff
@@ -0,0 +1,31 @@
+---
+  - name: check cluster status
+    command: kubectl cluster-info
+    register: cluster_status
+    ignore_errors: true
+
+  - block:
+
+    - name: Get node info
+      k8s_info:
+        kind: Infrastructure
+      register: infra_info
+
+    - name: Get kubectl version
+      command: kubectl version
+      register: kubectl_version
+
+    - name: get oc version
+      command: oc version
+      register: oc_version
+      ignore_errors: true
+
+    - name: Set cluster info as facts
+      set_fact:
+        stockpile_k8s_cluster_info:
+          cluster_name: "{{ infra_info.resources[0].status.infrastructureName }}"
+          platform: "{{ infra_info.resources[0].status.platform }}"
+          kubectl_version: "{{ kubectl_version.stdout_lines | default('') }}"
+          oc_version: "{{ oc_version.stdout_lines | default('') }}"
```

Comment:
Does it make sense to scrape the output of $ oc version for client, server and kube version info and set them as individual facts?